packages/forms-web-app/src/controllers/eligibility/planning-department.js

```diff
@@ -1,31 +1,67 @@
+const logger = require('../../lib/logger');
+const { getDepartmentFromName } = require('../../services/department.service');
+const { getDepartmentFromId } = require('../../services/department.service');
+const { createOrUpdateAppeal } = require('../../lib/appeals-api-wrapper');
 const { getDepartmentData } = require('../../services/department.service');
 const { VIEW } = require('../../lib/views');
 
 exports.getPlanningDepartment = async (req, res) => {
   const { departments } = await getDepartmentData();
-
+  const { appeal } = req.session;
+  let appealLPD = '';
+  if (appeal.lpaCode) {
+    const lpd = await getDepartmentFromId(appeal.lpaCode);
+    if (lpd) {
+      appealLPD = lpd.name;
+    }
+  }
```

Comment:
seems fair enough that because this code is needed in two controllers, we have some repetition here. If we need this in a third place, I think we should consider extracting to a middleware like we do for the `appeal`.

Reply:
Yep I agree and I wondered too but I think it will be the only times